### PR TITLE
Rename DM type parameter to Dm

### PR DESCRIPTION
**Pull request by JurajSadel**
closes https://github.com/esp-rs/esp-hal/issues/2775

### CODE REVIEW

**Review by bugadani — APPROVED** [review]
Thanks

**Comment by bjoernQ** [comment]
We have a "couple" of places where we call the type param just `M` or `MODE` (or `DmaMode`?) - should we use the opportunity to align these?

**Comment by Dominaezzz** [comment]
What does DM actually mean? DMA mode?

**Comment by bugadani** [comment]
I don't think it needs one

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by bugadani** [comment]
Thanks you! Please fix formatting, then feel free to merge this :)

File: esp-hal/src/touch.rs [hunk 1]
```diff
@@ -85,12 +85,12 @@ pub struct TouchConfig {
 }
 
 /// This struct marks a successfully initialized touch peripheral
-pub struct Touch<'d, TOUCHMODE: TouchMode, MODE: Mode> {
+pub struct Touch<'d, TOUCHMODE: TouchMode, Dm: Mode> {
```
Comment (jessebraham) [inline esp-hal/src/touch.rs:88]: Outside the scope of this PR, but we should probably search for other violations of this rule as well.

**Pull request merged**